build/v1/types.go

```diff
@@ -992,6 +997,8 @@ type WebHookTrigger struct {
 type ImageChangeTrigger struct {
 	// lastTriggeredImageID is used internally by the ImageChangeController to save last
 	// used image ID for build
+	// NOTE: this needs to be set only in the status, not spec, to avoid unnecessary triggering of builds
+	// when BuildConfig's are reapplied.
```

Comment:
Since this is `omitempty`, I don't see harm adding an `// +optional` tag.

Note should indicate that this field is ignored in `spec`.

Reply:
updates for this pushed @adambkaplan